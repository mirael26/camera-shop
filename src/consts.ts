export const AppUrl = {
  Main: '/',
  Catalog: '/catalog',
  Page: '/page_',
  Product: '/product',
  Cart: '/cart',
  NotFound: '/404',
  UnknownError: '/unknown_error',
  ServerUnavailable: '/server_unavailable',
} as const;

export const ApiUrl = {
  Promo: '/promo',
  Products: '/cameras',
  Reviews: '/reviews',
  Similar: '/similar',
  Promocode: '/coupons',
  Order: '/orders',
} as const;

export const Modal = {
  AddToCart: 'addToCartModal',
  DeleteFromCart: 'deleteFromCartModal',
  Review: 'reviewModal',
  OrderSuccess: 'orderSuccessModal',
} as const;

export const Tab = {
  Features: 'features',
  Description: 'description',
} as const;

export const Month = {
  0: 'января',
  1: 'февраля',
  2: 'марта',
  3: 'апреля',
  4: 'мая',
  5: 'июня',
  6: 'июля',
  7: 'августа',
  8: 'сентября',
  9: 'октября',
  10: 'ноября',
  11: 'декабря',
} as const;

export const Param = {
  Sort: 'sort',
  Order: 'order',
  Page: 'page',
  Tab: 'tab',
  PriceMin: 'price_min',
  PriceMax: 'price_max',
  Category: 'category',
  Level: 'level',
  Type: 'type',
} as const;

export const SortType = {
  Price: 'price',
  Rating: 'rating',
} as const;

export const SortOrder = {
  Asc: 'asc',
  Desc: 'desc',
} as const;

export const Category = {
  Camera: 'Фотоаппарат',
  VideoCamera: 'Видеокамера',
} as const;

export const Level = {
  Beginner: 'Нулевой',
  Amateur: 'Любительский',
  Professional: 'Профессиональный',
} as const;

export const CameraType = {
  Digital: 'Цифровая',
  Film: 'Плёночная',
  Instant: 'Моментальная',
  Collectible: 'Коллекционная',
} as const;
