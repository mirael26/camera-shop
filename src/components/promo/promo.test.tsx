import { fireEvent, getByText, screen } from '@testing-library/react';
import axios from 'axios';
import { AppUrl } from '../../consts';
import { renderTestApp } from '../../test/helpers/render-test-app';
import { renderWithReduxAndRouter } from '../../test/helpers/render-with-redux-and-router';
import { promoMock } from '../../test/mocks';
import Promo from './promo';

jest.mock('axios');

describe('Promo', () => {
  beforeEach(() => {
    jest.mocked(axios).get.mockResolvedValue(promoMock);
  });

  test('Renders correctly', () => {
    const promo = renderWithReduxAndRouter(<Promo/>, { initialState: { 
      data: { promo: promoMock }
    }});
    expect(promo).toMatchSnapshot();
  });

  test('Makes one get-query after render', () => {
    renderWithReduxAndRouter(<Promo/>);
    expect(jest.mocked(axios).get).toBeCalledTimes(1);
  });

  test('Renders right title', () => {
    renderWithReduxAndRouter(<Promo/>, { initialState: { 
        data: { promo: promoMock }
      }
    });
    const title = screen.getByTestId('promo-title');
    expect(title).toHaveTextContent(/Look 54/i);
  });

  test('Redirect to product', () => {
    renderTestApp(null, { route: AppUrl.Catalog,
      initialState: { 
        data: { promo: promoMock }
      }
    });
    const bunner = screen.getByTestId('banner');
    const productLink = getByText<HTMLLinkElement>(bunner, /Подробнее/i);

    expect(screen.queryByTestId('product-page')).not.toBeInTheDocument();
    fireEvent.click(productLink);
    expect(screen.getByTestId('product-page')).toBeInTheDocument();
  });
});
