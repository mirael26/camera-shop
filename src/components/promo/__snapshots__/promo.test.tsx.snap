// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Promo Renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="banner"
        data-testid="banner"
      >
        <picture>
          <source
            srcset="img/content/promo.webp, img/content/promo.webp 2x"
            type="image/webp"
          />
          <img
            alt="баннер"
            height="280"
            src="img/content/promo.jpg"
            srcset="img/content/promo@2x.jpg 2x"
            width="1280"
          />
        </picture>
        <p
          class="banner__info"
        >
          <span
            class="banner__message"
          >
            Новинка!
          </span>
          <span
            class="title title--h1"
            data-testid="promo-title"
          >
            Look 54
          </span>
          <span
            class="banner__text"
          >
            Профессиональная камера от известного производителя
          </span>
          <a
            class="btn"
            href="/catalog/product/7"
          >
            Подробнее
          </a>
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="banner"
      data-testid="banner"
    >
      <picture>
        <source
          srcset="img/content/promo.webp, img/content/promo.webp 2x"
          type="image/webp"
        />
        <img
          alt="баннер"
          height="280"
          src="img/content/promo.jpg"
          srcset="img/content/promo@2x.jpg 2x"
          width="1280"
        />
      </picture>
      <p
        class="banner__info"
      >
        <span
          class="banner__message"
        >
          Новинка!
        </span>
        <span
          class="title title--h1"
          data-testid="promo-title"
        >
          Look 54
        </span>
        <span
          class="banner__text"
        >
          Профессиональная камера от известного производителя
        </span>
        <a
          class="btn"
          href="/catalog/product/7"
        >
          Подробнее
        </a>
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
